packaging/packaging_test.go

```diff
@@ -0,0 +1,782 @@
+				testDirs.large = tmpDirMap
+			} else if directory.dirName == "small" {
+				testDirs.small = tmpDirMap
+			} else if directory.dirName == "moving" {
+				testDirs.moving = tmpDirMap
+			} else if directory.dirName == "empty" {
+				testDirs.empty = tmpDirMap
+			} else if directory.dirName == "restricted" {
+				testDirs.restricted = tmpDirMap
+			} else if directory.dirName == "restrictedEmpty" {
+				testDirs.restrictedEmpty = tmpDirMap
+			} else if directory.dirName == "split" {
+				testDirs.split = tmpDirMap
+			} else if directory.dirName == "tar" {
+				testDirs.tar = tmpDirMap
+			}
+		}
+	}
+	return nil
+}
+
+func shutdown() {
+	fmt.Println("Tearing down for packaging tests")
+	os.RemoveAll(testingDir)
+}
+
+func TestMain(m *testing.M) {
+	err := setup()
+	if err != nil {
+		fmt.Println("Can not run tests because setup failed to create testing directories and files")
+		shutdown()
+	} else {
+		code := m.Run()
+		shutdown()
+		os.Exit(code)
+	}
+}
+
+func TestNeedSplit(t *testing.T) {
+	// create the needSplitTests
+	needSplitTests := []struct {
+		name     string
+		fileList []os.FileInfo
+		maxBytes int64
+		want     bool
+	}{
+		{
+			name:     "test split required",
+			fileList: testDirs.large.files,
+			maxBytes: 1 * 1024 * 1024,
+			want:     true,
+		},
+		{
+			name:     "test split not required",
+			fileList: testDirs.small.files,
+			maxBytes: 100 * 1024 * 1024,
+			want:     false,
+		},
+	}
+	for _, tt := range needSplitTests {
+		// using tt.name from the case to use it as the `t.Run` test name
+		t.Run(tt.name, func(t *testing.T) {
+			testPackager.maxBytes = tt.maxBytes
+			got := testPackager.needSplit(tt.fileList)
+			if tt.want != got {
+				t.Errorf("Outcome for test %s:\nReceived: %v\nExpected: %v", tt.name, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBuildLocalCSVFileList(t *testing.T) {
+	// create the buildLocalCSVFileList tests
+	buildLocalCSVFileListTests := []struct {
+		name     string
+		dirName  string
+		fileList []os.FileInfo
+	}{
+		{
+			name:     "test regular dir",
+			dirName:  filepath.Join(testDirs.large.directory, "data"),
+			fileList: testDirs.large.files,
+		},
+		{
+			name:     "test empty dir",
+			dirName:  filepath.Join(testDirs.empty.directory, "data"),
+			fileList: testDirs.empty.files,
+		},
+	}
+	for _, tt := range buildLocalCSVFileListTests {
+		// using tt.name from the case to use it as the `t.Run` test name
+		t.Run(tt.name, func(t *testing.T) {
+			got := testPackager.buildLocalCSVFileList(tt.fileList, tt.dirName)
+			want := make(map[int]string)
+			fileInfoList, _ := ioutil.ReadDir(tt.dirName)
+			for idx, file := range fileInfoList {
+				// generate the expected file list
+				if strings.HasSuffix(file.Name(), ".csv") {
+					want[idx] = filepath.Join(tt.dirName, file.Name())
+				}
+			}
+			for index, fileName := range want {
+				if fileName != got[index] {
+					t.Errorf("Expected %s but got %s", fileName, got[index])
+				}
+			}
```

Comment:
Could simplify this one:
```suggestion
			if !reflect.DeepEqual(got, want) {
				t.Errorf("%s got\n\t%v\nwant\n\t%v", tt.name, got, want)
			}
```

Reply:
Done 👍